File: app/src/main/java/com/android3/siegertpclient/data/team/teamsource/teamLocal/TeamLocalDataSource.java
package com.android3.siegertpclient.data.team.teamsource.teamLocal;

import com.android3.siegertpclient.data.tournament.Tournament;
import com.android3.siegertpclient.data.tournament.tournamentsource.tournamentLocal.TournamentDao;

import org.jetbrains.annotations.NotNull;

import java.util.Random;
import java.util.regex.Pattern;

public class TeamLocalDataSource {


    final TeamDao teamDao;


    public TeamLocalDataSource(TeamDao teamDao) {
        this.teamDao = teamDao;
    }

    public boolean check(@NotNull String teamName) {
        boolean b1 = Pattern.matches("[a-zA-Z0-9]",teamName);

        if(b1)  {
            return true;
        }
        return false;
    }

    public void saveUser(@NotNull Team newteam) {
        teamDao.insertTeam(newteam);
    }

    @NotNull
    public String getId() {

        return createRandomStr(15);

    }


    public static String createRandomStr(int length){
        String str = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
        Random random = new Random();
        StringBuffer stringBuffer = new StringBuffer();
        for (int i = 0; i < length; i++) {
            int number = random.nextInt(62);
            stringBuffer.append(str.charAt(number));
        }
        return stringBuffer.toString();
    }


}

File: app/src/main/java/com/android3/siegertpclient/data/team/teamsource/teamLocal/TeamLocalDataSource.kt
package com.android3.siegertpclient.data.team.teamsource.teamLocal

import com.android3.siegertpclient.data.tournament.Tournament
import java.util.regex.Pattern
/**
 * Local data source of the team.
 */
class TeamLocalDataSource(val teamDao: TeamDao) {
    fun check(teamName: String): Boolean {
        val b1 = Pattern.matches("[a-zA-Z0-9]", teamName)
        return if (b1) {
            true
        } else false
    }

    fun saveTeam(newteam: Team) {
        teamDao.insertTeam(newteam)
    }
    fun getUser(){
        teamDao!!.allTeams
    }
    fun deleteUser(){
        teamDao!!.deleteAllTeam()
    }
    fun updateUser(team :Team){
        teamDao!!.updateTeam(team)
    }
}